tunnel/tunnel.go

```diff
@@ -0,0 +1,100 @@
+// Package tunnel contains helpers for a traffic tunneling implementation
+package tunnel
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+
+	"go.viam.com/rdk/logging"
+)
+
+// ReaderSenderLoop implements a loop that reads bytes from the reader passed in and sends those bytes
+// using sendFunc. The loop will exit for any error received or if the context errors.
+func ReaderSenderLoop(
+	ctx context.Context,
+	r io.Reader,
+	sendFunc func(buf []byte) error,
```

Comment:
~Did you try using an `io.Writer` here? The signature seems to largely match. And then I expect we could use `io.Copy`: https://pkg.go.dev/io#Copy~

Cheuk and I talked offline. We feel that might just move the for-loop around instead of getting rid of it. I'm happy to not worry about it.